src/streamlit_pydantic/ui_renderer.py

```diff
@@ -512,6 +526,22 @@ def _render_union_property(
             property, self._schema_references
         )
 
+        # special handling when there are instance values and a discriminator property
+        # to differentiate between object types
+        if property.get("init_value") and property.get("discriminator"):
```

Comment:
I think there is a bug somewhere with the union handling. For me, the union field instance demo is not working correctly (not showing the union object):

<img width="786" alt="image" src="https://user-images.githubusercontent.com/2852129/179371267-eb79f1aa-3d45-41b0-9e54-db829bc58c75.png">

Reply:
Sorry I wasn't aware of this! 

I've now pinned pydantic to >=1.9 as you've suggested below.